### PR TITLE
Fix #210 & #282 add .gitlab-ci.yml & bin/acceptance-tests.sh

### LINKED ISSUE #210: add .gitlab-ci.yaml for dev deployments
Will add deployments for stage & prod after dev pipeline is verified.

### PR DESCRIPTION
**Pull request by jgmize**
- ported from the snippets service
  - https://github.com/mozmeao/snippets-service/blob/master/.gitlab-ci.yml
  - https://github.com/mozmeao/snippets-service/blob/master/bin/acceptance-tests.sh
- #210 
- #282

### CODE REVIEW

**Comment by jgmize** [comment]
Working on the failure in https://gitlab.com/mozmeao/basket/-/jobs/269519919 due to missing coverage.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    - meao
+  script:
+    - ${DOCKER_RUN_CMD} bash -c './manage.py makemigrations  | grep "No changes detected"'
+
+run-flake8:
+  extends: .test-stage
+  tags:
+    - meao
+  script:
+    - ${DOCKER_RUN_CMD} flake8 snippets
```
Comment (jpetto) [inline .gitlab-ci.yml]: s/snippets/basket

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  stage: test
+  variables:
+    DOCKER_DEBUG: "False"
+    DOCKER_ALLOWED_HOSTS: "*"
+    DOCKER_SECRET_KEY: "foo"
+    DOCKER_DATABASE_URL: "sqlite:///"
+    DOCKER_SITE_URL: "http://localhost:8000"
+    DOCKER_CACHE_URL: "dummy://"
+    DOCKER_ENABLE_ADMIN: "True"
+    DOCKER_SECURE_SSL_REDIRECT: "False"
```
Comment (jpetto) [inline .gitlab-ci.yml]: should these match the values in [docker-compose.yaml](https://github.com/mozmeao/basket/blob/master/docker-compose.yml#L80-L86)?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  script:
+    - ${DOCKER_RUN_CMD} flake8 snippets
+
+run-unit-tests:
+  extends: .test-stage
+  tags:
+    - meao
+  coverage: '/TOTAL.+ ([0-9]{1,3}%)/'
+  script:
+    - ${DOCKER_RUN_CMD} bash -c 'coverage run ./manage.py test --parallel; coverage report'
```
Comment (jpetto) [inline .gitlab-ci.yml]: this differs from the test command in `run-tests.sh`:

`py.test --junitxml=test-results/test-results.xml basket`

File: bin/update_config.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+BIN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source $BIN_DIR/../docker/bin/set_git_env_vars.sh # sets DOCKER_IMAGE_TAG
+
+pushd $(mktemp -d)
+git clone --depth=1 -b ${CONFIG_BRANCH:=master} ${CONFIG_REPO:=github-mozmar-robot:mozmeao/basket-config} basket-config
+cd basket-config
+
+set -u
+for DEPLOYMENT in {clock-,donateworker-,fxaeventsworker-,fxaworker-,web-,worker-,}deploy.yaml; do
```
Comment (jpetto) [inline bin/update_config.sh]: as we're passing in multiple clusters in `.gitlab-ci.yml`, we'll need to add an outer loop here as we do [in snippets](https://github.com/mozmeao/snippets-service/blob/master/bin/update-config.sh#L11)

File: bin/acceptance-tests.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+for url in ${URLS[*]}
+do
+    check_http_code ${BASE_URL}${url}
+done
+
+# Check a page that throws 404. Not ideal but will surface 500s
+check_http_code ${BASE_URL}/foo 404
+
```
Comment (jpetto) [inline bin/acceptance-tests.sh]: known to be missing, but need to add checks for zero length content & empty json

File: .gitlab-ci.yml (hunk 3 above)
Comment (jgmize) [reply .gitlab-ci.yml]: We decided not to use junit xml for gitlab; this was used by a Jenkins plugin.

File: .gitlab-ci.yml (hunk 2 above)
Comment (jgmize) [reply .gitlab-ci.yml]: No, we don't want or need mysql & redis container dependencies for things like flake8

File: bin/acceptance-tests.sh (hunk 5 above)
Comment (jgmize) [reply bin/acceptance-tests.sh]: Will follow up on additional acceptance tests in https://github.com/mozmeao/basket/issues/293

File: jenkins/branches/master.yml [hunk 6]
```diff
@@ -3,10 +3,10 @@ regions:
   - oregon-b
   - frankfurt
 apps:
-  - basket-dev
+  # - basket-dev
```
Comment (jpetto) [inline jenkins/branches/master.yml]: only remaining question: do we want to stop dev deployments with this PR? or should we wait until we have `basket-config` successfully doing that deploy?

Comment (jgmize) [reply jenkins/branches/master.yml]: yes, let's go ahead and stop dev deployments with this PR, then we can iterate on a `.gitlab-ci.yml` in the basket-config repo that automatically deploys to dev.

**Pull request merged**